### ISSUE TITLE
Broken links in docs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5051]
|<img alt="techtonik's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| techtonik reported|
|-|-|
|Trac ID|trac#5051|
|Type|defect|
|Created|2011-04-15 09:31:43Z|
|Branch|https://github.com/twisted/twisted/tree/epytext-links-5051|

Here is the broken link to epytext doc -http://twistedmatrix.com/trac/browser/trunk/doc/core/development/policy/coding-standard.xhtml?format=raw

Patch is included, but there are more pages with broken links.

Attachments:

* [epytext.broken.link.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d66/d66886e86179731ffb7203da6f98961134d78b77/088abbb4387120b986db90a6811fa5f5f8b6658e.patch) (624 bytes) - added by techtonik on 2011-04-15 09:31:59Z - 
* [epytext.broken.link.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d66/d66886e86179731ffb7203da6f98961134d78b77/f6547fc592bc45736f0e44e8b3d67267dbe2d66b.patch) (624 bytes) - added by techtonik on 2011-04-15 09:31:59Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5051 5051
type__defect defect
reporter__techtonik techtonik
priority__normal normal
milestone__ 
branch__branches_epytext_links_5051 branches/epytext-links-5051
branch_author__thijs__techtonik thijs, techtonik
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1302859903000000 1302859903000000
changetime__1305214814000000 1305214814000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="techtonik's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|techtonik commented|
|-|-|

Other pages with invalid links reported by http://validator.w3.org/checklink are:
 * http://twistedmatrix.com/documents/current/core/howto/endpoints.html
 * http://twistedmatrix.com/documents/current/core/howto/choosing-reactor.html
 * http://twistedmatrix.com/documents/current/core/howto/glossary.html
 * http://twistedmatrix.com/documents/current/core/howto/cred.html
 * http://twistedmatrix.com/documents/current/core/howto/basics.html
 * http://twistedmatrix.com/documents/current/core/howto/process.html
 * http://twistedmatrix.com/documents/current/core/howto/rdbms.html
 * http://twistedmatrix.com/documents/current/core/howto/testing.html
 * http://twistedmatrix.com/documents/current/core/howto/tutorial/protocol.html

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [31696]) Branching to 'epytext-links-5051'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [31697]) apply epytext.broken.link.patch, refs [#5051](https://github.com/twisted/twisted/issues/5051)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Thanks, I applied the patch and will fix those other links as well.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [31698]) fix (api) links in howto documents, refs [#5051](https://github.com/twisted/twisted/issues/5051)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Links in the documents listed by techtonik were fixed.

One thing though; the Divmod links in [doc/core/howto/glossary.xhtml](doc/core/howto/glossary.xhtml) are all broken, should they be removed (except for the link text)?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [31699]) Add news file, refs [#5051](https://github.com/twisted/twisted/issues/5051)

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_9"></a>|@jonathanj set owner to @thijstriemstra|
|-|-|

Replying to [thijs](#note_7):
> One thing though; the Divmod links in [doc/core/howto/glossary.xhtml](doc/core/howto/glossary.xhtml) are all broken, should they be removed (except for the link text)?

What about linking to the [Launchpad Divmod project](https://launchpad.net/divmod.org) instead of removing the links?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [jonathanj](#note_9):
> Replying to [thijs](#note_7):
> > One thing though; the Divmod links in [doc/core/howto/glossary.xhtml](doc/core/howto/glossary.xhtml) are all broken, should they be removed (except for the link text)?
> 
> What about linking to the [Launchpad Divmod project](https://launchpad.net/divmod.org) instead of removing the links?

Sounds good.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra commented|
|-|-|

(In [31758]) add launchpad links for divmod projects. refs [#5051](https://github.com/twisted/twisted/issues/5051)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_13"></a>|@jonathanj set owner to @thijstriemstra|
|-|-|

I think this looks good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [31812]) Merge epytext-links-5051: Fix broken (API) links in core documentation.

Author: thijs, techtonik
Reviewer: jonathanj
Fixes: [#5051](https://github.com/twisted/twisted/issues/5051)
